tests/php/src/Internal/DependencyManagement/AbstractServiceProviderTests.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * AbstractServiceProviderTests class file.
+ *
+ * @package Automattic\WooCommerce\Tests\Internal\DependencyManagement
+ */
+
+namespace Automattic\WooCommerce\Tests\Internal\DependencyManagement;
+
+use Automattic\WooCommerce\Internal\DependencyManagement\AbstractServiceProvider;
+use Automattic\WooCommerce\Internal\DependencyManagement\ExtendedContainer;
+use Automattic\WooCommerce\Tests\Internal\DependencyManagement\ExampleClasses\ClassWithConstructorArgumentWithoutTypeHint;
+use Automattic\WooCommerce\Tests\Internal\DependencyManagement\ExampleClasses\ClassWithDependencies;
+use Automattic\WooCommerce\Tests\Internal\DependencyManagement\ExampleClasses\ClassWithPrivateConstructor;
+use Automattic\WooCommerce\Tests\Internal\DependencyManagement\ExampleClasses\DependencyClass;
+use League\Container\Definition\DefinitionInterface;
+
+/**
+ * Tests for AbstractServiceProvider.
+ */
+class AbstractServiceProviderTests extends \WC_Unit_Test_Case {
```

Comment:
Should we also have a test case for classes with scalar type hints? I wonder what our automatic argument resolution does in that case?

Reply:
Scalar type hints without default value should fail as we're not registering instances of scalars in the container, but it's a good idea to add a test for the failing case.